webpack/base.config.js

```diff
@@ -2,12 +2,17 @@ import path from 'path';
 import webpack from 'webpack';
 import ExtractTextPlugin from 'extract-text-webpack-plugin';
 import writeStats from './utils/write-stats';
+import config from 'config';
+import fs from 'fs';
 
 const Webpack_isomorphic_tools_plugin = require('webpack-isomorphic-tools/plugin');
 const webpack_isomorphic_tools_plugin =
     new Webpack_isomorphic_tools_plugin(require('./webpack-isotools-config'))
         .development();
 
+// https://github.com/lorenwest/node-config/wiki/Webpack-Usage
+fs.writeFileSync(path.resolve(__dirname, '../tmp/client.json'), JSON.stringify(config))
```

Comment:
I understand the need for this but it seems very strange to me that a configuration file has side effects. Possibly this could be moved to a build step?

Reply:
Were you thinking of something as simple as abstracting the current approach with the `fs.writeFileSync` side effect abstracted out into a plugin: https://gist.github.com/gl2748/9f7a05190ef742c3943df5aae8d9885d ?

Or something that leverages webpacks' virtual filesystem ...
https://github.com/rmarscher/virtual-module-webpack-plugin

Regardless -  

I do think this is alright for now. I understand the concern but in the interest of getting this merged think this is OK ... `node-config` at the time of writing is not compatible with webpack and a hack makes sense... The above approaches could be the basis for a re-visit...

Reply:
let's please get it right. there isn't some massive rush here.

Reply:
Ok.